### PR TITLE
DM-38452: Update with new DC2 templates and import scripts.

### PR DESCRIPTION
**Pull request by erinleighh**
Updated the preloaded repo with new templates generated in DM-
38451. Added import_templates.py and adjusted existing scripts to use it. Cleaned up remaining files.

### CODE REVIEW

File: scripts/import_templates.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+# Command-line options
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-b", dest="src_dir", default="/sdf/group/rubin/repo/dc2",
+                        help="Repo to import from, defaults to '/sdf/group/rubin/repo/dc2'.")
+    parser.add_argument("-t", dest="src_collection", required=True,
+                        help="Template collection to import from.")
+    parser.add_argument("-w", dest="where",
+                        help="Query datasets dimension expression.")
```
Comment (parejkoj) [inline scripts/import_templates.py:59]: Please specify what this is for: "... to restrict the templates that will be imported from the source collection."

**Review by parejkoj — APPROVED** [review]
One documentation comment, otherwise this looks good.

I think we definitely should revisit the question of whether and how to consolidate all these scripts, since I think they're all basically the same now. The `where` we added to this one should be added to the others, too, at which point why aren't we just putting them all somewhere together? Please bring this up at a standup or AP Pipelines meeting.

**Review by parejkoj — APPROVED** [review]

**Comment by parejkoj** [comment]
Hang on, I think the `scripts/import_templates.py` file got lost in the shuffle. You refer to it in `make_preloaded`, but I don't see it here?

File: scripts/import_templates.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-b", dest="src_dir", default="/sdf/group/rubin/repo/dc2",
+                        help="Repo to import from, defaults to '/sdf/group/rubin/repo/dc2'.")
+    parser.add_argument("-t", dest="src_collection", required=True,
+                        help="Template collection to import from.")
+    parser.add_argument("-w", dest="where",
+                        help="Query datasets dimension expression to restrict the templates \
+                              that will be imported from the source collection..")
```
Comment (parejkoj) [inline scripts/import_templates.py:60]: Extra period, and you can use string auto-concatenation.

```suggestion

                        help="Query datasets dimension expression to restrict the templates "
                              "that will be imported from the source collection.")
```

File: scripts/import_templates.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+lsst.log.configure_pylog_MDC("DEBUG", MDC_class=None)
+
+
+########################################
+# Command-line options
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-b", dest="src_dir", default="/sdf/group/rubin/repo/dc2",
+                        help="Repo to import from, defaults to '/sdf/group/rubin/repo/dc2'.")
```
Comment (parejkoj) [inline scripts/import_templates.py:55]: I think for this you can just use `/repo/dc2`.

https://developer.lsst.io/usdf/storage.html#butler-access

You also don't need to repeat the default in the docstring.

```suggestion
    parser.add_argument("-b", dest="src_dir", default="/repo/dc2",
                        help="Repo to import from.")
```

**Review by parejkoj — COMMENTED** [review]
Thanks for the missing file. Couple small changes.

Just to check: you did re-run the template generation code and make the templates on this ticket the newest ones?

File: scripts/import_templates.py (hunk 3 above)
Comment (erinleighh) [reply scripts/import_templates.py:55]: It throws an error when using just `/repo/dc2`. (Or at least it did when I was originally attempting to do this and the other ci template tickets.)

**Comment by erinleighh** [comment]
> Just to check: you did re-run the template generation code and make the templates on this ticket the newest ones?

Yes, I did a full rerun and all the preloaded content is new from the rerun.

**Pull request merged**